addons/survey/tests/test_survey_invite.py

```diff
@@ -258,3 +258,51 @@ def test_survey_invite_token_by_email_nosignup(self):
         answers = self.env['survey.user_input'].search([('survey_id', '=', self.survey.id)])
         self.assertEqual(len(answers), 1)
         self.assertEqual(answers.partner_id.display_name, first_partner.display_name)
+
+    @users('survey_user')
+    def test_survey_invite_with_template_attachment(self):
+        """
+        Test that a group_survey_user can send a survey that includes an attachment from the survey invite's
+            email template
+        """
+        mail_template = self.env['mail.template'].create({
+            'name': 'test mail template',
+            'attachment_ids': [Command.create({
+                'name': 'some_attachment.pdf',
+                'res_model': 'mail.template',
+                'datas': 'test',
+                'type': 'binary',
+            })],
+        })
+
+        self.user_survey = self.env['survey.survey'].with_user(self.survey_user).create({
+            'title': 'User Created Survey',
+            'access_mode': 'public',
+            'users_login_required': False,
+            'users_can_go_back': False,
+        })
+        self.env['survey.question'].with_user(self.survey_user).create({
+            'title': 'First page',
+            'survey_id': self.user_survey.id,
+            'sequence': 1,
+            'is_page': True,
+            'question_type': False,
+        })
+        self.env['survey.question'].with_user(self.survey_user).create({
+            'title': 'Test Free Text',
+            'survey_id': self.user_survey.id,
+            'sequence': 2,
+            'question_type': 'text_box',
+        })
```

Comment:
Can probably be embedded in survey creation using question_ids = [(0, 0, {...}), (0, 0, {...})] ?